File: backend/src/main/java/com/coing/domain/coin/market/controller/dto/MarketResponse.java
package com.coing.domain.coin.market.controller.dto;

import com.coing.domain.coin.market.dto.MarketResponseDto;

import jakarta.validation.constraints.NotNull;

public record MarketResponse(
	@NotNull
	String code,
	@NotNull
	String koreanName,
	@NotNull
	String englishName,
	Boolean isBookmarked
) {
	public static MarketResponse from(MarketResponseDto market) {
		return new MarketResponse(market.code(), market.koreanName(), market.englishName(), market.isBookmarked());
	}
}

File: backend/src/main/java/com/coing/domain/coin/market/controller/dto/MarketResponse.kt
package com.coing.domain.coin.market.controller.dto

import com.coing.domain.coin.market.dto.MarketResponseDto
import jakarta.validation.constraints.NotNull

data class MarketResponse(
	@field:NotNull
	val code: String,

	@field:NotNull
	val koreanName: String,

	@field:NotNull
	val englishName: String,

	val isBookmarked: Boolean
) {
	companion object {
		fun from(market: MarketResponseDto): MarketResponse {
			return MarketResponse(
				code = market.code,
				koreanName = market.koreanName,
				englishName = market.englishName,
				isBookmarked = market.isBookmarked
			)
		}
	}
}